### PR TITLE
Add user friend

### PR DESCRIPTION
**Pull request by brandonbothell**

### CODE REVIEW

File: src/entity/User/User.ts [hunk 1]
```diff
@@ -1,6 +1,14 @@
...
-import { UserSettings, UserVerification, UserLevel, UserBalance, UserProfile, UserReputation } from '.'
+import {
+  UserSettings,
+  UserVerification,
+  UserLevel,
+  UserBalance,
+  UserProfile,
+  UserReputation
+} from '.'
+import { UserFriend } from './UserFriend'
```
Comment (tgoins) [inline src/entity/User/User.ts]: Add this to class to the index.ts so you can combine this import on line 3.

File: src/entity/User/User.ts [hunk 2]
```diff
@@ -46,7 +58,10 @@ export class User {
   @OneToOne(type => UserReputation, userReputation => userReputation.user)
   reputation: UserReputation
 
-  constructor (user?: User) {
+  @OneToMany(type => UserFriend, userFriend => userFriend)
+  friendships: UserFriend
```
Comment (tgoins) [inline src/entity/User/User.ts]: Rename this to `friends`

Comment (tgoins) [reply src/entity/User/User.ts]: UserFriend should be `UserFriend[]`

File: src/entity/User/UserFriend.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  ManyToOne
+} from 'typeorm'
+import { User } from '.'
+
+@Index(['user', 'friend'], { unique: true })
+@Entity()
+export class UserFriend {
+  @PrimaryGeneratedColumn() id: number
+
+  @Index({ unique: true })
```
Comment (tgoins) [inline src/entity/User/UserFriend.ts]: remove the unique part. Should just be `@Index()`

File: src/entity/User/UserFriend.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+  @ManyToOne(type => User, user => user.friendships)
```
Comment (tgoins) [inline src/entity/User/UserFriend.ts]: `friendships` should be `friends`

File: src/entity/User/UserFriend.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  user: User
+
+  @Index({ unique: true })
+  @ManyToOne(type => User, user => user.friendships)
```
Comment (tgoins) [inline src/entity/User/UserFriend.ts]: `friendships` should be `friends`

File: src/entity/User/UserFriend.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+@Index(['user', 'friend'], { unique: true })
+@Entity()
+export class UserFriend {
+  @PrimaryGeneratedColumn() id: number
+
+  @Index({ unique: true })
+  @ManyToOne(type => User, user => user.friendships)
+  user: User
+
+  @Index({ unique: true })
```
Comment (tgoins) [inline src/entity/User/UserFriend.ts]: Remove the unique part. Should just be `@Index()`

**Review by tgoins — CHANGES_REQUESTED** [review]
Close. Just a few small things.

**Comment by brandonbothell** [comment]
@tannerz28 done

**Review by tgoins — APPROVED** [review]

**Pull request merged**